torch/_functorch/partitioners.py

```diff
@@ -741,13 +746,13 @@ def cleanup_recompute_tags(joint_module: fx.GraphModule) -> fx.GraphModule:
     non-recomputable to allow for that.
     """
     for node in joint_module.graph.nodes:
-        if must_recompute(node):
+        if prefer_recompute(node):
             for user in node.users:
                 if (
-                    must_recompute(user)
-                    and user.meta["recompute"] > node.meta["recompute"]
+                    prefer_recompute(user)
+                    and user.meta["ac_graph_id"] > node.meta["ac_graph_id"]
```

Comment:
is this an independent bugfix? Or is this part of the "partitioner should respect prefer_recompute" change.

Mostly a curiosity question: my understanding from the comment above is that if you run code like:
```
checkpoint_f = checkpoint(f)
checkpoint(g) = checkpoint(g)
out = f(g(inp))
```

Then AC requires us to save the inputs to f (outputs of g), but in our tag-based system every node would have the recompute tag: so you need some notion of "which AC subgraph does a node belong to" (`ac_graph_id`) to tell the partitioner that it should save the output of the first subgraph. Is that right?

Reply:
this is part of the "recompute" tag cleanup to disentangle its two meanings: 1) recompute policy, 2) AC subgraph ID, by splitting them into two tags.

yes your understanding is exactly right :)